### PR TITLE
MAISTRA-2370: Handle tombstones when handling events from the namespa…

### PR DESCRIPTION
**Pull request by knrc**
…ce specific caches

Handles tombstones when forwarding events from the namespace specific cache to the aggregated cache

### CODE REVIEW

File: pkg/listwatch/listinformer.go [hunk 1]
```diff
@@ -130,6 +130,10 @@ func (li *listerInformer) stop() {
 }
 
 func (li *listerInformer) newWatchEvent(eventType watch.EventType, obj interface{}) *watch.Event {
+	tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+	if ok {
+		obj = tombstone.Obj
+	}
 	return &watch.Event{
```
Comment (jwendell) [inline pkg/listwatch/listinformer.go]: Worth to add another check like 
```go
if _, ok := obj.(runtime.Object); !ok {
   print err; return nil
}
```
before trying to use it below?

**Review by jwendell — APPROVED** [review]

**Review by brian-avery — APPROVED** [review]

File: pkg/listwatch/listinformer.go (hunk 1 above)
Comment (knrc) [reply pkg/listwatch/listinformer.go]: Sounds like a good idea

Comment (knrc) [reply pkg/listwatch/listinformer.go]: @jwendell should be handled now

**Comment by knrc** [comment]
It looks as if something is up with prow, lots of deployments are timing out and not deploying.

**Comment by knrc** [comment]
Prow is not healthy, I'm going to merge this manually

**Comment by dgn** [comment]
I guess it doesn't act on closed PRs :(

**Pull request closed without merging**